spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractInstancePersistingDeployer.java

```diff
@@ -67,6 +68,7 @@ protected void beforeDelete(D definition) {
 	@Override
 	public void undeploy(String name) {
 		Assert.hasText(name, "name cannot be blank or null");
+		validateBeforeUndeploy(name);
```

Comment:
Same here, it seems that `validateBeforeUndeploy` is doing the exact same thing as the code below.